Form created. Added validation with RHF & yup

orchidenver opened the pull request:
During this session, I've separeted search input from MainPage.tsx into new component Form.tsx. In a new component I applied form validation for search input with react-hook-form and yup libraries.
Also some minor style improvements have been added in order to make the UI better.

Commit:
form created. added validation with rhf & yup

ivan-ivan commented on src/components/Form.tsx:
```diff
@@ -0,0 +1,126 @@
+
+export default function Form(props: IAppProps) {
+  const theme = useTheme();
+  const matches: boolean = useMediaQuery<string | undefined>(
+    theme.breakpoints.up("sm")
```
It might be better to give the variable more descriptive name like `isMobileResolution` or something like that. The name `matches` doesn't really reflect the essence of it.

Here's a good article on boolean variables naming https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig

ivan-ivan commented on src/components/Form.tsx:
```diff
@@ -0,0 +1,126 @@
+    resolver: yupResolver(schema),
+  });
+  const handleFormSubmit: SubmitHandler<FormData | Mode> = (values) => {
+    console.log(values);
```
It's considered a good practice to remove console logs before committing to master

ivan-ivan reviewed (changes requested):
Overall looks good. Just left 2 minor comments

orchidenver committed:
PR amendments

The name of the boolean variable changed. Also deleted all the unnecessary logs to the console.

ivan-ivan reviewed (approved):
LGTM
